docs/requirements/requirements.md

```diff
@@ -0,0 +1,105 @@
+# Hardware requirements of TwPM module
+
+[tcg_client_v13r27]:<https://trustedcomputinggroup.org/wp-content/uploads/TCG_PCClientTPMInterfaceSpecification_TIS__1-3_27_03212013.pdf>
+[tcg_client_PTP_v1p5r14]:<https://trustedcomputinggroup.org/wp-content/uploads/PC-Client-Specific-Platform-TPM-Profile-for-TPM-2p0-v1p05p_r14_pub.pdf>
+
+## Requirements
+
+### Power supply
+
+#### Power supply voltage
+
+* TPM MUST support a supply voltage of 1.8V or 3.3V.
+* TPM MAY support supply and I/O voltages of both 1.8V and 3.3V.
+* TPM MAY support other supply voltages.
+
+Source: [tcg_client_v13r27] lines 1373-1375
+
+#### Power supply max current
+
+Maximum power supply taken by TPM MUST be 250mA.
+
+Source: [tcg_client_PTP_v1p5r14], Table 60, VDD row
+
+### Memory requirements
+
+1. The TPM SHALL be capable of supporting a minimum of 100KB written
+to TPM non-volatile memory over the life of the TPM.
+2. The TPM shall support allocation of at least 68 indices, with a total
+minimum data size of 3834 Bytes(decimal).
+
+The TPM specification does not define the required lifetime or endurance
+of the NV memory.
+Source: [tcg_client_PTP_v1p5r14], Chapter 4.5.1
+
+### SPI Interface
+
+#### Frequency
+
+TPM must be capable of operating at frequencies 10-24MHz. However, the maximum
+frequency requirement may be raised in the future, so the target platform
+should have the SPI interface with capability to handle higher frequencies.
+
+Source: [tcg_client_v13r27] Chapter 6.4.1, line 1331
```

Comment:
@3mdeb-karolzmyslowski Specification is one thing. How about the **real** SPI frequencies we can meet on the mainboards - what SPI frequencies are provided by some selected chipsets, just to have a second reference? I specifically provided some Intel chipset datasheet we could use to have an idea on what we should expect in reality?

@krystian-hebel Do you agree it would be imporant to note this as well?

Reply:
Not sure about this. Chipsets often support more than one frequency. We would have to check which one is actually chosen by firmware and that is significantly more complicated than just finding some datasheet. It would be good to know this so we can test on as high frequency as possible, though.

Reply:
@miczyg1 ^

Reply:
Thanks for referencing. On Intel PCH systems, TPM SPI frequency is chosen in descriptor straps configured by FITc. Typically I saw 17MHz. But yeah we have to account for higher frequencies.

Reply:
I've taken specification for *Intel® 300 Series Chipset Family On-Package Platform Controller Hub (PCH) Datasheet* (link: https://www.intel.com/content/www/us/en/products/docs/chipsets/300-series-chipset-on-package-pch-datasheet-vol-1.html) There are few SPI interfaces, with various frequencies:
 * Table 10-23. SPI Timings (17 MHz) 
 * Table 10-26. SPI Timings (48 MHz)
 (and so on).
Is there any code of conduct which SPI interface is used for flash memory?

Reply:
There is one SPI interface for flash and TPM. But it uses different CS pins. CS0 and CS1 are used fro flash memory only, CS2 is for TPM use only.